src/coreclr/vm/ilstubcache.cpp

```diff
@@ -501,57 +484,38 @@ MethodDesc* ILStubCache::GetStubMethodDesc(
 
         if (SF_IsSharedStub(dwStubFlags))
         {
-            size_t cbSizeOfBlob = pParams->m_cbSizeOfBlob;
+            size_t cbSizeOfBlob = pHashBlob->m_cbSizeOfBlob;
             AllocMemHolder<ILStubHashBlob> pBlobHolder( m_heap->AllocMem(S_SIZE_T(cbSizeOfBlob)) );
 
             CrstHolder ch(&m_crst);
 
-            ILCHASHENTRY*   phe         = NULL;
-
-            bool bNew;
-            phe = (ILCHASHENTRY*)FindOrAdd((LPVOID)pParams, bNew);
-            bILStubCreator |= bNew;
-
-            if (NULL != phe)
+            ILStubCacheEntry* phe = m_hashMap.Lookup(pHashBlob);
+            if (phe == NULL)
             {
-                if (bNew)
-                {
-                    pBlobHolder.SuppressRelease();
-
-                    phe->m_pMethodDesc   = pMD;
-                    pBlob = pBlobHolder;
-                    phe->m_pBlob         = pBlob;
+                pBlobHolder.SuppressRelease();
+                pBlob = pBlobHolder;
 
-                    _ASSERTE(pParams->m_cbSizeOfBlob == cbSizeOfBlob);
-                    memcpy(pBlob, pParams, cbSizeOfBlob);
+                phe = new ILStubCacheEntry{ pMD, pBlob };
```

Comment:
This is a throwing `new` - is that okay in this code path?

Reply:
It should be ok - `m_hashMap.Add` is throwing too.